packages/dashboard/src/components/sidePanel/utils/index.ts

```diff
@@ -20,7 +21,14 @@ export const useInput: <T>(path: string) => [T, Dispatch<T>] = (path) => {
   const onInputValueChange = () => {
     const currentWidgetState = get(selectedWidget, path);
     if (inputValue !== currentWidgetState) {
-      // TODO: dispatch an action to update widget
+      const newWidget = set(cloneDeep(selectedWidget), path, inputValue);
+      const updateWidgetAction: UpdateWidgetsAction = {
```

Comment:
you can use `onUpdateWidgetsAction` to create this

Reply:
updated